packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -708,9 +712,32 @@ export class GarbageCollector implements IGarbageCollector {
         });
 
         /**
-         * Set up the initializer which initializes the state from the GC data in base snapshot.
+         * Set up the initializer which initializes the tombstone state from the data in base snapshot.
+         * This is done during container runtime load so that the tombstone state is updated before any objects
+         * are loaded or used.
          */
-        this.initializeStateFromBaseSnapshotP = new LazyPromise<void>(async () => {
+        this.initializeTombstoneStateFromBaseSnapshotP = new LazyPromise<void>(async () => {
```

Comment:
Why don't we move this logic to `initializeBaseState()`?

Reply:
We can. Then we will have to make baseSnapshotP a member of the class so they are equivalent IMO.

Reply:
I think it'll make the class a little easier to read, unless baseSnapshotP is used elsewhere

Reply:
good point. updated it.